rust/lance-encoding/src/encoder.rs

```diff
@@ -430,6 +475,83 @@ fn check_dict_encoding(arrays: &[ArrayRef], threshold: u64) -> bool {
     true
 }
 
+fn check_fixed_size_encoding(arrays: &[ArrayRef], byte_width: &mut u64) -> bool {
+    // check if all arrays in arrays have the same length
+    if arrays.is_empty() {
+        return false;
+    }
+
+    // make sure no array has an empty string
+    if !arrays.iter().all(|arr| {
+        if let Some(arr) = arr.as_string_opt::<i32>() {
+            arr.iter().flatten().all(|s| !s.is_empty())
+        } else if let Some(arr) = arr.as_binary_opt::<i32>() {
+            arr.iter().flatten().all(|s| !s.is_empty())
+        } else if let Some(arr) = arr.as_string_opt::<i64>() {
+            arr.iter().flatten().all(|s| !s.is_empty())
+        } else if let Some(arr) = arr.as_binary_opt::<i64>() {
+            arr.iter().flatten().all(|s| !s.is_empty())
+        } else {
+            panic!("wrong dtype");
+        }
+    }) {
+        return false;
+    }
```

Comment:
Why is this needed if you have a check later on to filter out lengths of zero anyways?  Ideally we only iterate though all values in the array a single time.

Reply:
We need to distinguish between null values and empty strings (both end up having offset lengths = 0). Since we want the fixed size encoding to support nulls, but we want to use the binary encoding for empty strings. So this check is to filter out the empty ones, and the one later handles the nulls.